src/garrysmod/gamemodes/zombiesurvival/gamemode/vgui/grid.lua

```diff
@@ -0,0 +1,144 @@
+local PANEL = {}
+
+AccessorFunc(PANEL, "horizontalMargin", "HorizontalMargin", FORCE_NUMBER)
+AccessorFunc(PANEL, "verticalMargin", "VerticalMargin", FORCE_NUMBER)
+AccessorFunc(PANEL, "rowHeight", "RowHeight", FORCE_NUMBER)
+
+function PANEL:Init()
+    self.cells = {}
+
+    self:SetPaintBackground(false)
+
+    self:Dock(TOP)
+end
+
+function PANEL:SetVerticalMargin(verticalMargin)
+    self.verticalMargin = verticalMargin
+    self:DockMargin(0, verticalMargin / 2, 0, verticalMargin / 2)
+end
+
+function PANEL:AddCell(cell)
+    table.insert(self.cells, cell)
+
+    cell:SetParent(self)
+
+    cell:Dock(LEFT)
+
+    cell.PerformLayout = function(self, _, _)
+        local parent = self:GetParent()
+
+        self:DockMargin(parent:GetHorizontalMargin() / 2, 0, parent:GetHorizontalMargin() / 2, 0)
+        -- (194 / 4 - 8) = 40.5, so there is a chance that this will result in a
+        -- floating point value that gets converted to an integer when running
+        -- SetWide(), causing things to be skewed by a pixel or 2.
+        -- parent:GetWide() % #parent.cells would get that remainder if you
+        -- wanted to do correction.
+        self:SetWide(parent:GetWide() / #parent.cells - parent:GetHorizontalMargin())
+    end
+
+    self:SetTall(self:CalculateHeight())
```

Comment:
The `Row` element is docked to the `TOP` which means it's width will automatically fill the parent. We set its height to the tallest element in the row. 

I recognize this won't catch things like a list where there may be many small elements stacked on top of each other (e.g. a scoreboard (;). That's where `SetRowHeight` comes in, to override the `CalculateHeight` function.